README.md

```diff
@@ -9,92 +9,388 @@
 
 ## Overview
 
-We can now get students thinking outside of the component box and in terms of
-somewhat larger applications. We can move from simply having handlers outside
-of our components to keeping state outside of components as well. We're not
-quite ready for full-blown flux, but we at least have the concepts of actions
-and stores, and we can already make use of them:
+In this lesson we're going to learn about a concept called "unidirectional"
+data-flow. Web Applications are primarily centered around rendering application
+state and data. It's extremely important to choose an architecture that allows
+us to scale our application as we add new components and functionality. Thinking
+about how our components, actions and state containers communicate is crucial
+in order to organize everything beyond trivial demos.
 
-```javascript
-// stores/Store.js
-module.exports = class Store {
-  constructor(state) {
-    this.listeners = []
-    this.state = state
+## Towards Unidirectional Data Flow
+
+### Using Change Listeners
+
+To understand what "unidirectional data flow" actually means, let's first
+consider a real-world example:
+
+We want to implement a simple application that allows us to organize tasks in
+some form of project board, similar to Trello:
+
+![Trello Screenshot](./assets/Trello.png)
+
+Trello allows you to raise issues, represented by "cards", assign them and move
+them into different columns, whereas each column represents a distinct step of
+your workflow.
+
+When create an application like that, it's always helpful to first think about
+how one would go about organizing the underlying data:
+
+Each card can be represented by a JSON object:
+
+```json
+{
+  "title": "Create Mockups",
+  "id": 123
+}
+```
+
+Each card can only ever be in exactly one column. Each column has a name and a
+distinct set of cards associated with it:
+
+```js
+{
+  "name": "TODO",
+  "id": 456,
+  "cards": [{
+    {
+      "title": "Create Mockups",
+      "id": 123
+    }
+  }]
+}
+```
+
+Now that we have a set of well-defined models and a beautifully designed UI, we
+can go about structuring our component hierarchy. In this case, the most basic
+screen is fairly simple: We have a project, which can be represented by a
+single, stateful component, multiple column and a variety of card components.
+
+In other words, this is how our project's render function could look like:
```

Comment:
'...this is **what** ... could look like' (or 'how ... could look', remove the 'like') — up to you